cmd/generate.go

```diff
@@ -0,0 +1,45 @@
+package cmd
+
+import (
+	"dynamic-buildkite-template/generator"
+	"os"
+
+	"github.com/spf13/cobra"
+)
+
+func init() {
+	// cobra.OnInitialize()
+	generateCmd.PersistentFlags().String("trivyPlugin", "v1.18.2", "provide trivy plugin version")
+	generateCmd.PersistentFlags().String("shellPlugin", "", "provide shell plugin version")
+	generateCmd.PersistentFlags().Bool("ignoreUnfixed", true, "provide if unfixed items are to be ignored")
+	generateCmd.PersistentFlags().String("skipFiles", "cosign.key", "provide files to be skipped in trivy plugin")
+	generateCmd.PersistentFlags().String("shellCheckFiles", "script.sh", "provide files to be checked by the shell plugin")
+
+	generateCmd.PersistentFlags().StringArray("severity", []string{"CRITICAL", "HIGH"}, "provide the severity")
+	generateCmd.PersistentFlags().StringArray("securityChecks", []string{"config", "secret", "vuln"}, "provide the security checks")
+}
+
+func Execute() error {
+	return generateCmd.Execute()
+}
+
+var generateCmd = &cobra.Command{
+	Use:   "generate",
+	Short: "Generates trivy step for the provided options",
+	Long: `
+Usage of dynamic-buildkite-template
+This Program generates trivy step for the provided options
+	`,
+	Run: func(cmd *cobra.Command, args []string) {
+		var g = generator.Generator{
+			TrivyPlugin:     mustGetStringFlag(cmd, "trivyPlugin"),
```

Comment:
this PR seems more geared towards trivy and not so much `dynamic`. If we were to make this dynamic i think the generator paramater just needs to be a map and update the templates that are generated to be based on the contents of the map. Each time we have to add a new feature we have to open up the code again instead of just adding in a template.

I think this needs to be updated to a `map[string]map[string]string` and the templates need to be updated to use the map.

a sample of template using a map.
```go
package main

import (
	"os"
	"text/template"
)

func main() {
	m := map[string]string{"foo!": "Bar!"}
	t := template.Must(template.New("").Parse(`Bar? {{index . "foo!"}}`))
	t.Execute(os.Stdout, m)
}

```

```
Bar? Bar!
Program exited.
```

https://go.dev/play/p/YuJX5A1CB2

Reply:
Thanks @angrieralien I will consider this suggestion in future PR's.